### PR TITLE
Add peer-to-peer network slides

### PR DESCRIPTION
**Pull request by matnad**
# Description

Slide set for Transaction Capacity, Peer-to-Peer Network.

## Type of Change

- [x] Major feature (New files or assets, large changes to files)

# Checklist:

- [x] I have performed a self-review of my own changes
- [x] I have made sure that any changed LaTeX files compile properly
- [x] I have assigned at least one reviewer

### CODE REVIEW

File: assets/figures/node_software.tex [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	
+	% Title
+%	\node[draw, align=center] at (2.2,4.5) {Peer-to-Peer};
+%	
+
+	
+	% Agents
+	\node (agenta) at (1,0) {\includegraphics[height = 2 cm]{../assets/images/agents/avatar_discover}};
+	\uncover<-1>{\node (agentc) at (10,0) {\includegraphics[height = 2 cm, decodearray={0.6 0 0.6 0 0.6 0}] {../assets/images/agents/avatar_rand4}};} % grayed out agent for first slide
+	\uncover<2->{\node (agentb) at (10,0) {\includegraphics[height = 2 cm] {../assets/images/agents/avatar_rand4}};}
```
Comment (golmit) [inline assets/figures/node_software.tex:15]: Use old avatars for consistency with other slide decks.

File: assets/figures/node_software.tex [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	
+	\node[above] at (agenta.north) {Discover};
+	
+	% Actions
+	\uncover<2->{\draw[<->, thick, dashed]	(agenta.east) -- node [above] {Connect} (agentb.west);}
+	\uncover<3->{\draw[->, thick, dashed]	(agenta.north east) -- node [above] {Share} (agentb.north west);}
+	\uncover<4->{\draw[<-, thick, dashed]	(agenta.south east) -- node [above] {Access} (agentb.south west);}
```
Comment (golmit) [inline assets/figures/node_software.tex:22]: Maybe increase the size of the agents that the upper and lower arrows are between them.

File: assets/figures/node_protocol.tex [hunk 3]
```diff
@@ -0,0 +1,16 @@
+
+\begin{tikzpicture}[scale = 0.67]
+	
+	% Agents
+	\node (agenta) at (1,0) {\includegraphics[height = 2 cm]{../assets/images/agents/avatar_rand1}};
+	\node (agentb) at (10,0) {\includegraphics[height = 2 cm] {../assets/images/agents/avatar_rand4}};
```
Comment (golmit) [inline assets/figures/node_protocol.tex:6]: Use the old avatars for consistency with other slide decks.

File: assets/figures/peer_discovery.tex [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+		\node (agentg2) at (agentg) {\includegraphics[width = 0.6 cm]{../assets/images/agents/avatar_rand4}};	
+		\node (agentd2) at (agentd) {\includegraphics[width = 0.6 cm]{../assets/images/agents/avatar_rand1}};	
+		
+		\draw[<->, thick, dashed]	(agentg.west) -- (agentc.east);
+		\draw[<->, thick, dashed]	(agentc.south) -- (agentd.north);
+		
+	}
+	
+	
+\end{tikzpicture}
```
Comment (golmit) [inline assets/figures/peer_discovery.tex:68]: Very clear animation!

File: assets/figures/node_software.tex (hunk 2 above)
Comment (matnad) [reply assets/figures/node_software.tex]: 👍 Done

File: assets/figures/peer_discovery.tex (hunk 4 above)
Comment (matnad) [reply assets/figures/peer_discovery.tex]: 🙏

**Review by Chainomics — COMMENTED** [review]
As discussed, all of the issues seem to be related to `decodearray`. Please remove and use multiple versions of the same icon. This is rather unfortunate and messy, but we cannot risk having a slide set that yields different results, depending on the student's pdf viewer. ;-)

File: slides/peer_to_peer.tex [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+			\input{../assets/figures/peer_discovery2}
+		\end{tikzpicture}
+	\end{center}
+	%\vspace{0.5 cm}
+	\uncover{Basic peer discovery:}
+	\begin{enumerate}
+		\item Connect to a known peer
+		\item Ask this peer for a list of their connections
+		\item Connect to a random set of peers from this list
+		\item<2-> Repeat steps 2 and 3 until enough connections are established
```
[the remark this answers is not in the document]
Comment (Chainomics) [reply slides/peer_to_peer.tex:167]: fixed.

File: slides/peer_to_peer.tex [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+		\item<3-> \color{focus}Further decentralizes \color{black}the system
+	\end{itemize}
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{Peer Discovery}
+	\begin{center}
+		\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
+			\input{../assets/figures/peer_discovery}
```
[the remark this answers is not in the document]
Comment (golmit) [reply slides/peer_to_peer.tex:142]: @Chainomics Depending on the typesetter the slide either compiles properly or not.

File: slides/peer_to_peer.tex [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+		\item<2-> Connect to a known peer
+		\item<3-> Ask this peer for a list of their connections
+	\end{enumerate}
+\end{frame}
+%%%
+
+\begin{frame}{Peer Discovery}
+	\begin{center}
+		\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
+			\input{../assets/figures/peer_discovery2}
```
[the remark this answers is not in the document]
Comment (golmit) [reply slides/peer_to_peer.tex:158]: @Chainomics Once again, depending on the typesetter the slide either compiles properly or not.

File: slides/peer_to_peer.tex (hunk 6 above)
[the remark this answers is not in the document]
Comment (matnad) [reply slides/peer_to_peer.tex:142]: Replaced decodearray with static images, does that fix the problem @golmit

**Review by Chainomics — APPROVED** [review]
Works like a charm in a large variety of pdf viewers! :-) Thanks @matnad, glad we were able to figure out what caused the problem.

**Pull request merged**